### PR TITLE
fix: Implements handling of the high limb during fixed base scalar multiplication

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

Continuation of https://github.com/noir-lang/acvm/issues/531 to implement logic to handle the high limb

## Problem\*

Resolves 

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: blackbox_solver/Cargo.toml [hunk 1]
```diff
@@ -31,17 +31,19 @@ p256 = { version = "0.11.0", features = [
...
+num-bigint.workspace = true
 
-# Barretenberg WASM dependencies
 rust-embed = { version = "6.6.0", features = [
     "debug-embed",
     "interpolate-folder-path",
     "include-exclude",
-] }
-
+] } # Barretenberg WASM dependencies
```
Comment (TomAFrench) [inline blackbox_solver/Cargo.toml:41]: Looks like this got mangled.

File: blackbox_solver/src/barretenberg/wasm/scalar_mul.rs [hunk 2]
```diff
@@ -14,12 +17,46 @@ impl ScalarMul for Barretenberg {
...
+            return Err(Error::FromFeature(FeatureError::InvalidGrumpkinScalarLimb {
+                limb_as_hex: hex::encode(high_bytes),
+            }));
+        }
+
+        let low_16_bytes = low_bytes[16..32].to_vec();
+        let high_16_bytes = high_bytes[16..32].to_vec();
+
+        let mut bytes = high_16_bytes.to_vec();
+        bytes.extend_from_slice(&low_16_bytes);
```
Comment (TomAFrench) [inline blackbox_solver/src/barretenberg/wasm/scalar_mul.rs:44]:
```suggestion
        let low_16_bytes = &low_bytes[16..32];
        let high_16_bytes = &high_bytes[16..32];

        let mut bytes = high_16_bytes.to_vec();
        bytes.extend_from_slice(low_16_bytes);
```

File: blackbox_solver/src/barretenberg/wasm/scalar_mul.rs [hunk 3]
```diff
@@ -14,12 +17,46 @@ impl ScalarMul for Barretenberg {
...
         low: &FieldElement,
-        _high: &FieldElement,
+        high: &FieldElement,
     ) -> Result<(FieldElement, FieldElement), Error> {
         let lhs_ptr: usize = 0;
         let result_ptr: usize = lhs_ptr + FIELD_BYTES;
-        self.transfer_to_heap(&low.to_be_bytes(), lhs_ptr);
 
+        let low_bytes = low.to_be_bytes();
+        let high_bytes = high.to_be_bytes();
```
Comment (TomAFrench) [inline blackbox_solver/src/barretenberg/wasm/scalar_mul.rs:26]: Why not use `try_into_u128` here? This would make readability of this function much nicer.

**Review by TomAFrench — APPROVED** [review]

File: blackbox_solver/src/barretenberg/wasm/scalar_mul.rs (hunk 3 above)
Comment (kevaundray) [reply blackbox_solver/src/barretenberg/wasm/scalar_mul.rs]: That method is doing a lot of allocations, so I decided to not go that route. Happy to go with `try_into_u128` for now as this method is mostly temporary

**Pull request merged**